plugins/out_opensearch/os_conf.c

```diff
@@ -111,10 +111,11 @@ struct flb_opensearch *flb_os_conf_create(struct flb_output_instance *ins,
         /* Check if the index has been set in the configuration */
         if (ctx->index) {
             /* do we have a record accessor pattern ? */
-            if (strchr(ctx->index, '$')) {
+            if (strchr(ctx->index, '$') != NULL) {
```

Comment:
Isn't this supposed  to be `if (ctx->index[0] == '$')` to avoid false positives?

Reply:
No, I can form an index like foo-$record['key']